File: BEST-App-main/lib/model/message.dart
class Message {
  final int id;
  final int userId;
  final int messageSampleId;
  final String dateSent;
  final String dateRead;
  final String content;

  Message(this.id, this.userId, this.messageSampleId, this.dateSent, this.dateRead, this.content);

  static Map<String, dynamic> toJson(Message ob) {
    return {
      'id': ob.id,
      'user_id': ob.userId,
      'message_sample_id': ob.messageSampleId,
      'date_sent': ob.dateSent,
      'date_read': ob.dateRead,
      'content': ob.content
    };
  }
}

File: spring-best/src/main/java/com/best/springbest/model/Message.java
package com.best.springbest.model;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "messages")
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "user_id",nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "message_sample_id",nullable = false)
    private MessageSample messageSample;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date date_sent;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date date_read;

    private String content;
}
